### ISSUE TITLE
Deployment Issue for Multi-Root project

### ISSUE THREAD

**Issue by gntpet** [microsoft/AL-Go#228]
Hey @freddydk, 

I have these projects/apps stored in multi-root repository

project **SRS.Core**, whic has 4 apps

- SRS.Core
- email-queue
- dynamic-layout
- import-salary-journal

project SRS.it, which has single app
- SRS.it

project SRS.no, which has single app
- SRS.no

Here is the depdency graph between these apps:
![image](https://user-images.githubusercontent.com/16342128/194535567-4d792839-a05a-4492-b3a1-c27e1d00d0d3.png)

if I specify **SRS.it** in PROJECTS secret, deployment code nicely finds dependencies and publishes 4 apps: _SRS.it, SRS.core,  email-queue, dynamic-layout apps_

But, I also want to deploy app _import-salary-journal_, so i change my secret to : **SRS.it, SRS.core** (since i must specify 'projects' here)

Now, deployment codes sees duplicate apps and fails with an error
![image](https://user-images.githubusercontent.com/16342128/194536176-ccf612ab-cb88-4bf1-a6ea-59db87621aad.png)

Is it a bug, or is this limitation on purpose?
What was the reason behind using 'PROJECTS' secret, instead of 'APPS' secret? Now, I specify list of projects to deploy, instead of list of APPS to deploy. Wonder what is the idea behind? I find it a bit limited, since I'm forced to deploy all apps from the project.

Best Regards, 
Gintautas

**Comment by gntpet** [comment]
I will contact you through Yammer

**Comment by freddydk** [comment]
I don't think you ever did, did you?

**Comment by freddydk** [comment]
Got it, will try to repro

**Comment by freddydk** [comment]
I have fixed the first issue in BcContainerHelper - and it should be in preview in approx. 1 hour.
I have added the request of using project/app to the backlog as a request, but the original idea is that projects should be shippable units, meaning that you would have apps together with dependent apps in one project (or in one repo if using single project repos).
Hence the artifacts - you always get all apps from a project in one .zip file together.
